osmaxx/profile/views.py

```diff
@@ -1,20 +1,148 @@
+from django.core.exceptions import ObjectDoesNotExist
+from django.core.mail import send_mail
 from django.core.urlresolvers import reverse
+from django.http import HttpResponseRedirect
+from django.shortcuts import redirect
+from django.template.defaultfilters import urlencode
+from django.utils.translation import ugettext as _
 from django.views import generic
 
 from osmaxx.contrib.auth.frontend_permissions import LoginRequiredMixin
 from osmaxx.profile.forms import ProfileForm
+from osmaxx.profile.models import Profile
 
 
-class ProfileView(LoginRequiredMixin, generic.UpdateView):
-    model = get_user_model()
+class SendVerificationEmailMixin(object):
+    RATE_LIMIT_SECONDS = 30
+
+    def _send_email_verification(self, profile):
+        if cache.get(profile.associated_user.id):
+            return
+        to_email = profile.unverified_email
+        if to_email:
+            cache.set(profile.associated_user.id, 'dummy value', timeout=self.RATE_LIMIT_SECONDS)
+            user_administrator_email = settings.OSMAXX['ACCOUNT_MANAGER_EMAIL']
+            token = profile.activation_key()
+            subject = _('Verify your email address')
+            message = _('In order to verify your email-address click the following link:\n'
+                        '{}?token={}'
+                        '\n'
+                        'If it wasn\'t you, just ignore this email.\n'
+                        .format(self.request.build_absolute_uri(reverse('profile:activation')), urlencode(token)))
+            send_mail(
+                subject=subject,
+                message=message,
+                from_email=user_administrator_email,
+                recipient_list=[to_email],
+            )
+            messages.add_message(
+                self.request, messages.INFO, _('To activate your email, click the link in the confirmation email.')
+            )
+
+
+class ProfileView(SendVerificationEmailMixin, LoginRequiredMixin, generic.UpdateView):
     form_class = ProfileForm
     template_name = 'profile/profile_edit.html'
 
+    def get(self, *args, **kwargs):
+        profile = self._get_or_create_profile()
+        user = self.request.user
+        if not (user.email or user.profile.unverified_email):
+            messages.add_message(
+                self.request,
+                messages.WARNING,
+                _('You have not set an email. You must set a valid email to use OSMaxx.')
+            )
+        if self.is_new_user():
+            self._move_email_from_user_to_profile(user, profile)
+            self._send_email_verification(profile)
+        else:
+            self._ensure_profile_has_email(profile, user)
+        user.refresh_from_db()
+        return super().get(*args, **kwargs)
+
+    def is_new_user(self):
+        user = self.request.user
+        return not (
+            user.profile.unverified_email or user.groups.filter(name=settings.OSMAXX_FRONTEND_USER_GROUP).exists()
+        )
```

Comment:
Extracting `user.profile.unverified_email is not None` and `user.groups.filter(name=settings.OSMAXX_FRONTEND_USER_GROUP).exists()` to local variables would allow to indicate what these intermediate values imply without the need for comments:
```python
    def is_new_user(self):
        user = self.request.user
        profile_already_existed = user.profile.unverified_email is not None
        user_already_had_access = user.groups.filter(name=settings.OSMAXX_FRONTEND_USER_GROUP).exists()
        return not (profile_already_existed or user_already_had_access)
```
eliminating the double negation:
```python
    def is_new_user(self):
        user = self.request.user
        profile_is_new = user.profile.unverified_email is None
        user_already_had_access = user.groups.filter(name=settings.OSMAXX_FRONTEND_USER_GROUP).exists()
        return profile_is_new and not user_already_had_access
```